src/components/CandidateForm.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react'
+import styles from './CandidateForm.module.css'
+
+const CandidateForm = (props) => {
+    let defaultState = Object.assign({}, ...props.inputData.map(input => {
+        const {value, name} = input;
+        return {
+            [name]: value
+        }
+    }));
+    const [state, setState] = useState(defaultState);
+
+    const handleOnChange = (e) => {
+        e.preventDefault();
+        const {name, value} = e.target;
+        setState(prevState => {
+            return {
+                ...prevState,
+                [name]: value
+            }
+        });
+    };
+
+    const formInputs = props.inputData.map(input => {
```

Comment:
A good tip is to name your functions based on what they do or return. I like to add _render to the functions that render elements. 

So here you would call this _renderFormInputs, when I see only formInputs I expect this function to return some form inputs data.